### ISSUE TITLE
Compiler crash (not ICE): Unknown exception during compilation: vector::_M_range_check: __n (which is 0) >= this->size() (which is 0) ```

### ISSUE THREAD

**Issue by agroce** [ethereum/solidity#10553]
## Description

This contract:
```solidity
interface ERC165 {
    /// @return `true` if the contract implements `interfaceID` and
    function supportsInterface(bytes4 interfaceID) external view returns (bool);
}
abstract contract ERC165MappingImplementation is ERC165 {
    function supportsInterface(bytes4 interfaceID) external view  {    }
}
interface Simpson {}
contract Lisa is ERC165MappingImplementation, Simpson {
}
```

produces:

```
Unknown exception during compilation: vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
```

when compiled with solc.

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer.

## Environment

- Compiler version:  0.7.6-develop.2020.12.9+commit.052b97e2.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

**Comment by agroce** [comment]
Confirmed this is not whatever is going on in #10532 -- on mac OS fresh checkout and build this contract produces:

```
Unknown exception during compilation: vector
```

**Comment by leonardoalt** [comment]
I can confirm this one:
```
(gdb) bt
#0  0x00007ffff6e53072 in __cxxabiv1::__cxa_throw (obj=obj@entry=0x55555646bc40, tinfo=0x7ffff6f7f268 <typeinfo for std::out_of_range>, 
    dest=0x7ffff6e68fa0 <std::out_of_range::~out_of_range()>) at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_throw.cc:78
#1  0x00007ffff6e495ed in std::__throw_out_of_range_fmt (__fmt=<optimized out>) at /build/gcc/src/gcc/libstdc++-v3/src/c++11/functexcept.cc:82
#2  0x00005555557405d6 in std::vector<std::shared_ptr<solidity::frontend::VariableDeclaration>, std::allocator<std::shared_ptr<solidity::frontend::VariableDeclaration> > >::_M_range_check (
    this=0x55555645c620, __n=0) at /usr/include/c++/10.2.0/bits/stl_vector.h:1073
#3  0x0000555555740169 in std::vector<std::shared_ptr<solidity::frontend::VariableDeclaration>, std::allocator<std::shared_ptr<solidity::frontend::VariableDeclaration> > >::at (
    this=0x55555645c620, __n=0) at /usr/include/c++/10.2.0/bits/stl_vector.h:1112
#4  0x000055555581d4de in (anonymous namespace)::copyMissingTags (_baseFunctions=std::set with 1 element = {...}, _target=..., _declaration=0x55555645ae60)
    at solidity/libsolidity/analysis/DocStringAnalyser.cpp:69
```

**Comment by agroce** [comment]
Oh good.  I was pretty sure if it happened on mac too it was real

**Comment by bshastry** [comment]
This looks like a duplicate of https://github.com/ethereum/solidity/issues/10433 to be fixed by https://github.com/ethereum/solidity/pull/10455

**Comment by agroce** [comment]
Somewhat different structure/failure, right (I get a segfault for #10433 vs the vector-crash here)?  Same fix, though, I assume (it's not quite an empty tag if I understand)

**Comment by bshastry** [comment]
@agroce To be precise, I think this issue is a duplicate of https://github.com/ethereum/solidity/issues/10433#issuecomment-739850148 (please let me know if that triggers a different failure for you).

In any case, bug collisions like this are quite likely given that fuzzers are operating in similar environments. Also, sometimes different failures are triggered for the same bug that may only be known post triage.

**Comment by bshastry** [comment]
Thanks for reporting the issue @agroce 

I am going to close this issue since the fix for #10433 has now been merged. Please reopen if you believe otherwise.
